python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -578,7 +653,15 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         butlerQC.put(result, outputRefs)
 
     @timeMethod
-    def run(self, *, exposures, id_generator=None, result=None):
+    def run(
+        self,
+        *,
+        exposures,
+        id_generator=None,
+        result=None,
+        background_flat=None,
+        illumination_correction=None,
+    ):
```

Comment:
Just hanging this thread at a semi-arbitrary location: can we have this do a check of the input image metadata, to look for something ISR could put there, to confirm that it has been dome-flattened only?

Reply:
I've added appropriate metadata for the isr tasks (which really should have been there all along), and check it here.  At the moment I'm raising a `RuntimeError` but I am not sure if that's appropriate.  (But it might be because it's a crashworthy programming mistake).

Reply:
I think `RuntimeError` is okay.  I think it's not an `AlgorithmError`.  It might be a `lsst.pipe.base.FatalQuantumError`, which means "stop everything, the configuration and/or inputs are bad!".